tests/irdl/test_attr_constraint.py

```diff
@@ -153,3 +161,29 @@ class NoParamAttr(BaseNoParamAttr): ...
 def test_constraint_repr(constr: AttrConstraint, expected: str):
     assert repr(constr) == expected
     assert eval(repr(constr)) == constr
+
+
+@pytest.mark.parametrize(
+    "input, output",
+    [
+        (TensorType(i32, [2, 2]), MemRefType(i32, [2, 2])),
+        (UnrankedTensorType(i32), UnrankedMemRefType.from_type(i32)),
+    ],
+)
+def test_tensor_to_memref(
+    input: TensorType | UnrankedTensorType, output: MemRefType | UnrankedMemRefType
+):
+    assert TensorFromMemRefConstraint.tensor_to_memref(input) == output
+
+
+@pytest.mark.parametrize(
+    "input, output",
+    [
+        (MemRefType(i32, [2, 2]), TensorType(i32, [2, 2])),
+        (UnrankedMemRefType.from_type(i32), UnrankedTensorType(i32)),
```

Comment:
@superlopuh I don't really understand all the `memory_space` stuff, is there any case worth adding here?

Reply:
Oh yeah, that's actually pretty crucial, and easily broken

Reply:
I expect it's currently broken on main, but maybe this is a good chance to fix it

Reply:
I've just looked it up, tensor types don't have a memory space, so there's not much to do, I guess this should now be OK